### ISSUE TITLE
Step 2. AI model

### ISSUE THREAD

**Issue by escs34** [kookmin-sw/2019-cap1-2019_3#7]
스토리 :
-(사용자) "나"는 이전에 개발한 영상을 화면에 띄우는 프로그램에서, 띄우는 이미지에 대한 감정 분석을 하려고 한다.
-openCV로 사진에서 얼굴을 잘라낸후
-CNN 모델을 사용해 분석한 감정을 출력한다.

**Comment by escs34** [comment]
![Mini-Xception](https://user-images.githubusercontent.com/28584128/56125406-d347e600-5fb3-11e9-91c2-06aff572b7dc.png)

사용할 모델은 위와 같습니다. Xception architecture의 영향을 받아 seperable convolution을 사용하고, (1,1) 크기의 convolution 작업을 거친 결과를 residual module로 사용한 Mini-Xception model입니다.

가장 큰 특징은 매개변수의 갯수가 같은 성능을 낸 모델에 비해 100분의 1개 라는 것으로, 웹 서비스를 제공하는 과제 특성상 전방계산도 빠른 점이 도움이 될 것입니다.

Reference : https://appliedmachinelearning.blog/2018/11/28/demonstration-of-facial-emotion-recognition-on-real-time-video-using-cnn-python-keras/

**Comment by escs34** [comment]
![step2](https://user-images.githubusercontent.com/28584128/56125821-c4adfe80-5fb4-11e9-9391-116370d51254.png)

-사용할 CNN 모델에 따라 변경, 추가된 Class diagram 입니다.
-댓글 모듈에서 사용하는 모델이 pytorch로 구현되어 있기 때문에 추후에 tensorflow 에서 pytorch로 코드를 바꿔도 쓸 수 있도록 추상클래스 oracle을 경첩으로 두었습니다.
-학습 코드는 프로그램 코드에 포함시키지 않고,  jupyter notebook 상에서 학습 후 모델만 따로 저장할 것입니다.

**Comment by escs34** [comment]
학습결과

총 110번의 Epoch를 거쳤으며 validation set 정확도가 0.66~0.68 사이에 위치하였습니다.
Epoch 1/110
897/897 [============================>.] - ETA: 0s - loss: 1.7668 - acc: 0.3317
Epoch 00001: val_loss improved from inf to 1.69885, saving model to models/_mini_XCEPTION.01-0.37.hdf5
Epoch 2/110
896/897 [============================>.] - ETA: 0s - loss: 1.5072 - acc: 0.4349
Epoch 00002: val_loss improved from 1.69885 to 1.36495, saving model to models/_mini_XCEPTION.02-0.50.hdf5

.
.
.

Epoch 60/110
896/897 [============================>.] - ETA: 0s - loss: 0.8985 - acc: 0.6669
Epoch 00060: val_loss did not improve
Epoch 61/110
896/897 [============================>.] - ETA: 0s - loss: 0.8920 - acc: 0.6689
Epoch 00061: val_loss improved from 0.94373 to 0.94187, saving model to models/_mini_XCEPTION.61-0.65.hdf5

.
.
.

Epoch 109/110
897/897 [============================>.] - ETA: 0s - loss: 0.8444 - acc: 0.6856
Epoch 00109: val_loss improved from 0.93376 to 0.93361, saving model to models/_mini_XCEPTION.109-0.66.hdf5
Epoch 110/110
897/897 [============================>.] - ETA: 0s - loss: 0.8478 - acc: 0.6816
Epoch 00110: val_loss did not improve


[학습 결과.txt](https://github.com/kookmin-sw/2019-cap1-2019_3/files/3079681/default.txt)
